marshaler.go

```diff
@@ -319,39 +349,57 @@ func (enc *Encoder) encodeTableHeader(b []byte, key []string) ([]byte, error) {
 }
 
 func (enc *Encoder) encodeKey(b []byte, k string) ([]byte, error) {
+	needsQuotation, cannotUseLiteral, err := quotationNotLiteral(k)
+	if err != nil {
+		return nil, fmt.Errorf("encodeKey: %w", err)
+	}
+
+	switch {
+	case cannotUseLiteral:
+		return enc.encodeQuotedString(false, b, k), nil
+	case needsQuotation:
+		return enc.encodeLiteralString(b, k), nil
+	default:
+		return enc.encodeUnquotedKey(b, k), nil
+	}
+}
+
+var errTomlNoMultiline = errors.New("TOML does not support multiline keys")
+
+//nolint:cyclop
```

Comment:
I'm wondering if this linter (and the one that complains about function length) are really that useful here. I know the conventional wisdom says "write short functions", but I also find that splitting into many small functions sometimes just make the code harder to read. No need to change, just FYI. I might take a pass when you're done to see if there are some cyclo/length settings that I like better.